packages/opentelemetry-cloud-monitoring-exporter/test/monitoring.test.ts

```diff
@@ -135,6 +135,53 @@ describe('MetricExporter', () => {
       assert.strictEqual(result.code, ExportResultCode.FAILED);
     });
 
+    it('should return FAILED if project id promise is rejected', async () => {
+      await exporter['_projectId'];
+      exporter['_projectId'] = Promise.reject({
+        message: 'Failed to resolve projectId',
+      });
+
+      const result = await new Promise<ExportResult>(resolve => {
+        exporter.export([], result => {
+          resolve(result);
+        });
+      });
+
+      assert.deepStrictEqual(result, {
+        code: ExportResultCode.FAILED,
+        error: {
+          message: 'Await projectId failed: Failed to resolve projectId',
+        },
+      });
+    });
+
+    it('MetricsExporter#export should not raise an UnhandledPromiseRejectionEvent if projectId rejects', async () => {
+      const meter = new MeterProvider().getMeter('test-meter');
+      const labels: Labels = {['keya']: 'value1', ['keyb']: 'value2'};
+      const counter = meter.createCounter('name');
+      counter.add(10, labels);
+      await meter.collect();
+      const records = meter.getProcessor().checkPointSet();
+
+      await exporter['_projectId'];
+      exporter['_projectId'] = Promise.reject({
+        message: 'Failed to resolve projectId',
+      });
+
+      let unhandledPromiseRejectionEvent = false;
+      process.on('unhandledRejection', error => {
+        unhandledPromiseRejectionEvent = true;
+      });
+
+      const result = await new Promise<ExportResult>(resolve => {
+        exporter.export(records, result => {
+          resolve(result);
+        });
+      });
+
+      assert.strictEqual(unhandledPromiseRejectionEvent, false);
```

Comment:
Did you find this test failed before the fix? I'm not sure if `unhandledRejection` handler is guaranteed to be called before this assertion is made

Reply:
I wanted to make sure there was a regression test covering what we're improving. I ran this without my changes and it triggered a failure. Admittedly, I haven't used this in practice so I can't say if this is a good practice or not.